python-checks/src/main/java/org/sonar/python/checks/ModifiedParameterValueCheck.java

```diff
@@ -160,18 +196,19 @@ private static Map<Tree, String> getMutations(Parameter parameter) {
       return Collections.emptyMap();
     }
     Map<Tree, String> mutations = new HashMap<>();
-    for (Usage usage : paramSymbol.get().usages()) {
+    for (Usage usage : paramSymbol.usages()) {
       getKindOfWriteUsage(paramSymbol, defaultValueType, typeMutatingMethods, usage).ifPresent(s -> mutations.put(usage.tree().parent(), s));
     }
     return mutations;
   }
 
-  private static Optional<String> getKindOfWriteUsage(Optional<Symbol> paramSymbol, @Nullable String defaultValueType, Set<String> typeMutatingMethods, Usage usage) {
+  private static Optional<String> getKindOfWriteUsage(Symbol paramSymbol, @Nullable String defaultValueType, Set<String> typeMutatingMethods, Usage usage) {
     Tree parent = usage.tree().parent();
     if (parent.is(QUALIFIED_EXPR)) {
       QualifiedExpression qualifiedExpression = (QualifiedExpression) parent;
-      return getSymbolFromTree(qualifiedExpression.qualifier()).equals(paramSymbol) && isMutatingMethod(typeMutatingMethods, qualifiedExpression.name().name()) ?
-        Optional.of(MODIFIED_SECONDARY) : Optional.empty();
+
+      return getSymbolFromTree(qualifiedExpression.qualifier()).filter(paramSymbol::equals).isPresent()
+        && isMutatingMethod(typeMutatingMethods, qualifiedExpression.name().name()) ? Optional.of(MODIFIED_SECONDARY) : Optional.empty();
```

Comment:
These conditions are not covered. I cannot imagine any use case where one of these conditions could be false. However, I don't want to change the logic of the check with this PR.